Group 7

Nathandloria opened the pull request:
Completed last 10 bullets of "Software Development with Python" section. Provided detailed descriptions for each level in all 10 sub-sections.

noorbuchi commented:
Please add a detailed description of your changes and update your brach with `main`

enpuyou commented on assessment.md:
```diff
@@ -40,43 +40,130 @@ Replace TODOs with abbreviations, or emoji, or ...
+
+ *Good = Having high code coverage on a fairly regular basis and knowing how to further improve code coverage.
+
+ *Excellent = Always having code coverage, ensuring maximum software efficiency.
```
This level doesn't seem consistent with the previous ones. It would be good if you could provide the actual expected code coverage percentages of each level. Also, code coverage technically does not "ensure efficiency". It is just a measure of the percentage of the code that is being tested.

enpuyou commented on assessment.md:
```diff
@@ -40,43 +40,130 @@ Replace TODOs with abbreviations, or emoji, or ...
 * Reporting code coverage through an online provider
 
-  TODO: add description for each assessment level for this skill.
+ *None = Not knowing what code coverage is or how to use online provider to report code coverage.
```
This section needs to address "online provider", such as `codecov`.

enpuyou commented on assessment.md:
```diff
@@ -40,43 +40,130 @@ Replace TODOs with abbreviations, or emoji, or ...
+programmer is doing for most of the program
+
+ *Excellent  Utilizes clear and concise naming conventions to convey
+what the programmer is doing throughout the whole program
```
I think it is important to point out that students are expected to apply and adhere to a naming convention, for example, the use of the snake case in python.

enpuyou commented on assessment.md:
```diff
@@ -40,43 +40,130 @@ Replace TODOs with abbreviations, or emoji, or ...
-TODO: add description for each assessment level for this skill.
+ *None = Does not use exception handling in Python
+
+ *Inadequate = Using "try" and "except" for Exception Handling partially for the program.
```
Since try and except are actual python reserved keywords, use `try` instead of "try".

enpuyou replied:
No need to capitalize exception handling.

enpuyou commented on assessment.md:
```diff
@@ -40,43 +40,130 @@ Replace TODOs with abbreviations, or emoji, or ...
+
+ *Good = Having codes that are always properly debugged.
+
+ *Excellent = Having codes that are properly debugged as well as knowledge and application of multiple debugging methods.
```
Look into [logging](https://realpython.com/python-logging/)

enpuyou reviewed (changes requested):
Hi @Nathandloria, I have attached some of the reviews below. Overall, the PR needs to conform to the emoji and indentation format that are used throughout the other sections. I have updated your branch with `main` so you can take a look at the rest of the document. Additionally, try to stay consistent with the use of period at the end of each assessment. Thanks for working on this!

nathandloria commented:
> Hi @Nathandloria, I have attached some of the reviews below. Overall, the PR needs to conform to the emoji and indentation format that are used throughout the other sections. I have updated your branch with `main` so you can take a look at the rest of the document. Additionally, try to stay consistent with the use of period at the end of each assessment. Thanks for working on this!

Thank you for the reviews! We will get working on that. I just updated the file assessment.md to conform to the emoji and indentation expectations of the main branch.

noorbuchi commented on assessment.md:
```diff
@@ -56,43 +56,79 @@ Replace TODOs with abbreviations, or emoji, or ...
+
+  * :blush: = Having high code coverage on a fairly regular basis and knowing how to further improve code coverage.
+
+  * :laughing: = Always having code coverage, ensuring maximum software efficiency.
```
Since this is a bulleted list, I don't think that there should be spaces between each line. Please make sure to stay consistent with the main branch and throughout the file.

noorbuchi commented on assessment.md:
```diff
@@ -56,43 +56,79 @@ Replace TODOs with abbreviations, or emoji, or ...
-  TODO: add description for each assessment level for this skill.
+  * :flushed: = Does not attempt to implement any test features with pytest.
+
+  * :worried: = Low code coverage due to inefficient coding.
```
To clarify, code coverage is based on what percentage of code is getting tested in the test suite. It has little to do with the efficiency of the code being tested. I think you should clarify the meaning of code coverage in all assessment criteria

noorbuchi commented on assessment.md:
```diff
@@ -56,43 +56,79 @@ Replace TODOs with abbreviations, or emoji, or ...
+
+  * :neutral_face: = Knowing how to report code coverage and report at a fairly regular basis.
+
+  * :blush: =  Reporting code coverage at a regular basis and making attempts to improve code coverage through efficient coding.
+
```
Replacement suggestion, instead of `efficient coding` a better description would be `using thorough test suite`
